src/nft_swapper/Core.mo

```diff
@@ -44,11 +67,13 @@ module {
               parties = [caller];
             },
           );
+          // if (not (await submitCheck(caller, plan))) { return false };
           true;
         };
         case (?planStates) {
           switch (planStates.current) {
             case (#submit(submit)) {
+              // if (not (await submitCheck(caller, plan))) { return false };
```

Comment:
@rvanasa Logically, the check should go here, but when I try that, the unit tests will hang in different ways.

I'm not sure yet what's happening.

Reply:
I just took a look at this, and the CI issues were caused by `await async {}` working differently in the interpreter, causing the first `while` loop to run infinitely without reaching the other async blocks. Pushed a fix.

Reply:
Thanks for figuring this out.

I think that this was necessary indicates that the interpreter is not quite implementing `await` faithfully, though.